### DISCUSSION TITLE
What is the meaning of this piece of code?

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#320] [General]
`@handler : (GObject, Base) ~ Nil;`

https://github.com/objeck/objeck-lang/blob/c9f45bae3e36eb0fa6515d7e137b9a3940fa307b/core/lib/experimental/gtk/bindings/gtk3/gtk3_bindings.obs#L600

**Comment by ghost** [comment]
@objeck What does `(GObject, Base)` mean? `handler` has two data types? This poc is confusing!

**Reply by objeck** [reply]
It's the callback handler for a GTK event. [Here's](https://github.com/objeck/objeck-lang/blob/c9f45bae3e36eb0fa6515d7e137b9a3940fa307b/core/lib/experimental/gtk/bindings/gtk3/gtk3_bindings.obs#L628) an example.

**Comment by objeck** [comment]
Hi @iahung2, it has been months since I have looked at the generated code. I would start with [this ](https://github.com/objeck/objeck-lang/tree/master/core/lib/experimental/gtk/poc)demo, which worked on Linux and Windows. The demo was used as a guide for code generation.

**Comment by objeck** [comment]
It may be picked up if needed. As time permits, I want to create an ML framework based on TensorFlow or MXNet.

**Comment by ghost** [comment]
Is `@handler : (GObject, Base) ~ Nil; ` valid Objeck code? What does it mean? Which data type is `(GObject, Base)`?

**Reply by ghost** [reply]
@objeck Noted that `~` on Objeck is used for the return type of methods and functions. I think this syntax is one of the obscure feature of Objeck called `alias`. Please correct me if I'm wrong.

**Reply by objeck** [reply]
You are correct. The ``alias`` keyword is used to reference complex functional types. Here's an [example](https://github.com/objeck/objeck-lang/blob/ecd3af4f24f0b74b816e864e6ee9d563d7f23dbe/programs/deploy/lambda_17.obs#L13).
